### DISCUSSION TITLE
是否可以为form添加一层初始化的转换器和提交时的转换器

### DISCUSSION THREAD

**Discussion by xhh0223** [ant-design/ant-design#41427] [Ideas]
经常碰到的场景是
从后端获取到数据往往需要自己转换一次，
并且提交的时候还要从form收集到的值再转化为后端需要的值一次

如果这个form被多处地方使用，则会重复的编写转换逻辑，虽然我们可以把这个转换逻辑提取出来，
但是这个转换逻辑实际上与form的相关性高一点，应该和form放在一起，更加内聚一点。
这样就可以达到我填入的输入值，然后得到我想要的输出。

Frequently encountered scenarios are
Obtaining data from the backend often requires self conversion,
And when submitting, it is necessary to convert the values collected from the form into the values required by the backend once again
If this form is used in multiple places, the conversion logic will be repeatedly written, although we can extract the conversion logic,
However, this transformation logic is actually a bit more relevant to the form, and should be placed together with the form to be more cohesive.
This allows me to reach the input value I filled in, and then get the output I want.

**Comment by afc163** [comment]
Check `normalize`

**Reply by xhh0223** [reply]
normalize我在debugger的过程中发现是值改变后触发的

**Reply by afc163** [reply]
可以试试 getValueFromEvent

**Reply by xhh0223** [reply]
这个和normalize差不多，区别一个是事件。
我的目的是需要一个初始化的转换器，转换为form需要的值
提交后再给一个转换器，转换为我需要的值

**Reply by xhh0223** [reply]
不需要重复的去编写这个转换的过程，让这个转换的过程与form绑定在一起，这个是一个可选的回调方法

**Reply by guoyunhe** [reply]
normalize 只有输出的时候有效，如果要输入/输出都有效，那么应该还有一个对应的回调方法才对。

**Comment by crazyair** [comment]
可以用 getValueProps + getValueFromEvent 处理值的吞吐，参考 https://zhuanlan.zhihu.com/p/375753910
这里讲了将 RangePicker 数组转成表单 start、end 2个字段，另起其他任何转换都可以实现。

**Reply by xhh0223** [reply]
你这个封装显得太复杂了，我的目的很简单，就是在后端给我值再到我给form初始值之间加一个转换钩子，然后再给form提交时到我给后端值之前，加一层转行钩子。这个钩子应该是针对form进行封装的。不是formitem

**Reply by crazyair** [reply]
封装个 item，再封装几个  format 方法，日后使用非常爽的。麻烦是一时的，快乐是一世的。
